contracts/twap/CumulativeTwap.sol

```diff
@@ -11,70 +11,67 @@ contract CumulativeTwap is BlockContext {
     //
     // STRUCT
     //
+
     struct Observation {
         uint256 price;
         uint256 priceCumulative;
         uint256 timestamp;
     }
 
-    //
-    // EVENT
-    //
-    event PriceUpdated(uint256 price, uint256 timestamp, uint8 indexAt);
-
     //
     // STATE
     //
+
+    uint8 public currentObservationIndex;
     // let's use 15 mins and 1 hr twap as example
-    // if the price is being updated 15 secs, then needs 60 and 240 historical data for 15mins and 1hr twap.
+    // if the price is updated every 15 secs, then we need 60 and 240 historical data for 15mins and 1hr twap
     Observation[256] public observations;
 
-    uint8 public currentObservationIndex;
+    //
+    // INTERNAL
+    //
 
     function _update(uint256 price, uint256 lastUpdatedTimestamp) internal {
```

Comment:
Shouldn't `CachedTwap` inherit `IPriceFeedUpdate` to be compatible with `PriceFeddUpdater`?